tt_metal/impl/device/device.hpp

```diff
@@ -102,68 +93,54 @@ class Device {
 
     bool is_initialized() const { return this->initialized_; }
 
+    // Next group methods is proxying a call to Cluster
+    // Some methods can be replaced with ::get_soc_description
+    // dram_size_perf_channel is only used in tests
+    // dram_grid_size is only used in a test and reshard
     int num_dram_channels() const;
-
     uint32_t l1_size_per_core() const;
     uint32_t dram_size_per_channel() const;
-
     CoreCoord grid_size() const;
-
     CoreCoord logical_grid_size() const;
-
-    CoreCoord compute_with_storage_grid_size() const;
-
     CoreCoord dram_grid_size() const;
-
     CoreType core_type_from_virtual_core(const CoreCoord& virtual_coord) const;
     // Given a Virtual coordinate in noc_index space, get the equivalent coordinate in Virtual NOC0 space
     CoreCoord virtual_noc_coordinate(uint8_t noc_index, CoreCoord coord) const;
     // Given a coordinate in Virtual NOC0 Space, get the equivalent coordinate in Virtual noc_index space
     CoreCoord virtual_noc0_coordinate(uint8_t noc_index, CoreCoord coord) const;
+
     std::vector<CoreCoord> worker_cores_from_logical_cores(const std::vector<CoreCoord> &logical_cores) const;
     std::vector<CoreCoord> ethernet_cores_from_logical_cores(const std::vector<CoreCoord> &logical_cores) const;
     std::vector<CoreCoord> get_optimal_dram_bank_to_logical_worker_assignment();
 
     CoreCoord virtual_core_from_logical_core(const CoreCoord &logical_coord, const CoreType& core_type) const;
-
     CoreCoord worker_core_from_logical_core(const CoreCoord &logical_core) const;
 
     // Ethernet API
     CoreCoord ethernet_core_from_logical_core(const CoreCoord &logical_core) const;
     CoreCoord logical_core_from_ethernet_core(const CoreCoord &ethernet_core) const;
-
-    std::unordered_set<chip_id_t> get_ethernet_connected_device_ids() const {
-        return tt::Cluster::instance().get_ethernet_connected_device_ids(this->id_);
-    }
```

Comment:
There are a number of these kind of methods that just reach into `tt::Cluster::instance()` to do the lookup. We should be able to strip these methods from Device.

Reply:
Yes. There are also 32 allocator related methods in device